chrome/content/zotero/xpcom/schema.js

```diff
@@ -477,26 +479,37 @@ Zotero.Schema = new function(){
 			return deferred.promise;
 		})
 		.then(function (updated) {
-			if (runRemoteUpdateWhenComplete) {
-				var deferred = Q.defer();
-				if (updated) {
-					if (Zotero.Prefs.get('automaticScraperUpdates')) {
-						Zotero.proxyAuthComplete
-						.then(function () {
-							Zotero.Schema.updateFromRepository(2, function () deferred.resolve());
-						})
-					}
-				}
-				else {
-					Zotero.proxyAuthComplete
+			if (!runRemoteUpdateWhenComplete) return;
```

Comment:
I know this way is much cleaner, but this is all rewritten on api_syncing, so this will just cause conflicts. Anything that's not strictly necessary should stay unchanged. (4.0 should be considered close to frozen at this point.)